### ISSUE TITLE
gnome 41 support

### ISSUE THREAD

**Issue by rudrab** [micheleg/dash-to-dock#1524]
Hello developers,
Unlike many others, ewlsh's fork was working for me in gnome 40 (fedora 34). 
But in gnome 41, fedora 35, its not working. I have changed metadata.json to 41, still, I am getting an error symbol. Everything else is fine, but intellihide, with mouse gesture to unhide the dash is not working atall.

**Comment by ronidee** [comment]
Not working at all under **Debian bookworm/**sid, GNOME **40.4**, **Wayland**.

**Comment by vsv** [comment]
I've made ewlsh's fork working for me by replacing AppIconMenu with AppMenu in the codebase:

[ewlsh-gnome-41.patch.txt](https://github.com/micheleg/dash-to-dock/files/7270408/ewlsh-gnome-41.patch.txt)

**Comment by ronidee** [comment]
Sorry, apparently I'm kinda in the dark about this. What exactly is ewlsh's fork? Does his/her fork have more features, or just higher stability?
Can I apply this patch also to this fork? 

Thanks a lot for sharing your solution though! I will try this out tomorrow.

**Comment by vsv** [comment]
By "ewlsh's fork" I mean this pull request: https://github.com/micheleg/dash-to-dock/pull/1402 . For the majority of the users, it makes dash-to-dock work with Gnome 40; there are some bugs, however, which prevent it from being merged yet.

I've also discovered another fork of that fork, which solves the issues specifically with Gnome 41: https://github.com/frantisekz/dash-to-dock/tree/fzatlouk/gnome-41 . So if you need to get dash-to-dack working asap and cannot wait until everything is merged and released properly, the best course of action is the following:
1. Please ensure that you have `sassc` installed
2. Grab the sources of the fork and install the extension manually:
```
git clone https://github.com/frantisekz/dash-to-dock.git
cd dash-to-dock
git switch fzatlouk/gnome-41
make
make install
```

**Comment by maxborowsky** [comment]
Thx, @vsv

> I've made ewlsh's fork working for me by replacing AppIconMenu with AppMenu in the codebase:
> 
> [ewlsh-gnome-41.patch.txt](https://github.com/micheleg/dash-to-dock/files/7270408/ewlsh-gnome-41.patch.txt)

DTD works for me in GNOME 41 (Fedora 35 beta) with this patch, but right-clicking now does not call up the standard menus for application icons in the dock.

**Comment by slacka** [comment]
The `patch.txt` works for me in GNOME 41 on openSUSE Tumbleweed. Been running for a few days now. The only major issue is lack of right-click on the dock apps.

**Comment by maxborowsky** [comment]
Yeah, fork https://github.com/frantisekz/dash-to-dock/tree/fzatlouk/gnome-41 works perfectly. I don't see any bugs yet.

**Comment by erick-alvz** [comment]
version 70 is not working now Help !!

**Comment by riffer33** [comment]
erick-alvz - Gnome seems to automatically update extensions now, at least on my solus install, so it might have "updated" to the latest version on the extension website.  ergo the fzatlouk version I was using is gone.  I'm just looking into stopping the automatic updating now.

**Comment by erick-alvz** [comment]
@riffer33 did you manage to stop the automatic update?

**Comment by vsv** [comment]
> At present, I can't find a way to cancel the automatic update

You can "block" automatic updates by bumping "version" field to 71 in  "~/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/metadata.json"
